gensim/utils.py

```diff
@@ -906,10 +906,12 @@ def pickle(obj, fname, protocol=2):
 
 def unpickle(fname):
     """Load pickled object from `fname`"""
-    with smart_open(fname) as f:
+    with smart_open(fname, 'rb') as f:
         # Because of loading from S3 load can't be used (missing readline in smart_open)
-        return _pickle.loads(f.read())
-
+        if sys.version_info > (3, 0):
+            return _pickle.load(f, encoding='latin1')
```

Comment:
This works as a fix for when loading objects in Python 3 which were pickled in Python 2, which gives an exception.
Basically, Python 3 attempts to convert the pickled py2 object into a str object, when we need it to be bytes and gives an exception. I used the latin1 encoding for as a work around for that. [Asked on Stackoverflow](http://webcache.googleusercontent.com/search?q=cache:lHZB47tGMV4J:stackoverflow.duapp.com/questions/33596082/load-gensim-word2vec-computed-in-python-2-in-python-3+&cd=5&hl=en&ct=clnk&gl=in)